modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -2606,13 +2608,7 @@ private void updateScrollBarsAndCells(boolean recreate) {
             }
 
             lengthBar.setMax(1);
-            if (numCellsVisibleOnScreen == 0 && cellCount == 1) {
-                // special case to help resolve RT-17701 and the case where we have
-                // only a single row and it is bigger than the viewport
-                lengthBar.setVisibleAmount(flowLength / sumCellLength);
-            } else {
-                lengthBar.setVisibleAmount(viewportLength / estimatedSize);
-            }
+            lengthBar.setVisibleAmount( numCellsVisibleOnScreen / (float) cellCount);
```

Comment:
I'm concerned with the change to no longer use the estimated size. This will produce very different results for tables with variable row hights. @johanvos will very likely want to comment on this.

Also, The cast is unnecessary.

Minor: remove the space after the `(`

Reply:
I am very sorry at this point. I have made a mistake here because my changes were based on the code I had seen in 2017 and reported in JDK-8173321. At that time, the size calculation of the ScrollBar was based on the integer value of numCellsVisibleOnScreen.
I will revert the changes to VirtualFlow.